addons/project/tests/test_project_recurrence.py

```diff
@@ -44,543 +41,62 @@ def test_recurrence_simple(self):
-            repeat_on_month='date',
-            repeat_on_year=False,
-            weekdays=False,
-            repeat_day=31,
-            repeat_week=False,
-            repeat_month=False,
-            count=5)
-
-        self.assertEqual(len(dates), 6)
-        self.assertEqual(dates[0], date(2022, 11, 30))
-        self.assertEqual(dates[1], date(2023, 2, 28))
-        self.assertEqual(dates[2], date(2023, 5, 31))
-        self.assertEqual(dates[3], date(2023, 8, 31))
-        self.assertEqual(dates[4], date(2023, 11, 30))
-        self.assertEqual(dates[5], date(2024, 2, 29))
-
-    def test_recurrence_next_dates_year(self):
-        dates = self.env['project.task.recurrence']._get_next_recurring_dates(
-            date_start=date(2020, 12, 1),
-            repeat_interval=1,
-            repeat_unit='year',
-            repeat_type='until',
-            repeat_until=datetime(2026, 1, 1),
-            repeat_on_month=False,
-            repeat_on_year='date',
-            weekdays=False,
-            repeat_day=31,
-            repeat_week=False,
-            repeat_month='november',
-            count=10)
-
-        self.assertEqual(len(dates), 5)
-        self.assertEqual(dates[0], datetime(2021, 11, 30))
-        self.assertEqual(dates[1], datetime(2022, 11, 30))
-        self.assertEqual(dates[2], datetime(2023, 11, 30))
-        self.assertEqual(dates[3], datetime(2024, 11, 30))
-        self.assertEqual(dates[4], datetime(2025, 11, 30))
-
-    def test_change_subtask_of_recurrent_task(self):
-        with freeze_time(self.date_01_01):
-            task = self.env['project.task'].create({
-                'name': 'recurring task',
-                'project_id': self.project_recurring.id,
-                'recurring_task': True,
-                'repeat_interval': 2,
-                'repeat_unit': 'week',
-                'repeat_type': 'after',
-                'repeat_number': 2,
-                'mon': True,
-                'child_ids': [
-                    Command.create({'name': 'subtask'})
-                ],
-            })
-            recurrence = task.recurrence_id
-            task.child_ids.write({
-                'name': 'subtask 1',
-                'recurrence_update': 'future',
-            })
-
-        with freeze_time(self.date_01_15):
-            self.env['project.task.recurrence']._cron_create_recurring_tasks()
-            child = recurrence.task_ids.sorted('create_date', reverse=True)[0].child_ids
-            self.assertEqual(child.name, 'subtask 1')
-            child.write({
-                'name': 'subtask 2',
-                'recurrence_update': 'this',
-            })
-
-        with freeze_time(self.date_02_01):
-            self.env['project.task.recurrence']._cron_create_recurring_tasks()
-            child = recurrence.task_ids.sorted('create_date', reverse=True)[0].child_ids
-            self.assertEqual(child.name, 'subtask 1')
+            with Form(self.env['project.task']) as form:
```

Comment:
Since we need the result after the edition in the form view, we could do this instead:
```suggestion
            form = Form(self.env['project.task'])
```
and also review the tabulation since we will no longer have the with expression.

I suggest that because normally, when you use `form.save()`, we are outside the `with` and so the variable should not be accessible (I know Python authorise it but since we could avoid it why not doing it 😁 )